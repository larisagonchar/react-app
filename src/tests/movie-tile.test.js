import { render, screen } from '@testing-library/react';
import MovieTile from '../components/movie-tile/movie-tile';
import userEvent from '@testing-library/user-event';

describe('MovieTile component', () => {
  const movie = {
    genres: ['Action', 'Adventure'],
    release_date: 2004,
    title: 'Pulp Fiction',
    poster_path: ''
  };

  test('render movie tile', () => {
    const { asFragment } = render(<MovieTile movie={movie} />);

    expect(asFragment).toMatchSnapshot();
  });

  test('call onClick when user click on item', () => {
    const onItemClickMock = jest.fn();
    render(<MovieTile movie={movie} onClick={onItemClickMock} />);

    const container = screen.getByTestId('movie-tile');

    userEvent.click(container);

    expect(onItemClickMock).toHaveBeenCalledWith(movie);
  });
});