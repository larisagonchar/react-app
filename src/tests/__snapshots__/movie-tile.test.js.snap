// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile component render movie tile 1`] = `[Function]`;
